python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -235,6 +237,17 @@ def __post_init__(self) -> None:
                 # Need to do some file introspection to get the number
                 # of columns so that column projection works right.
                 raise NotImplementedError("Reading CSV without header")
+        elif self.typ == "ndjson":
+            # TODO: consider handling the low memory option here
+            # (maybe use chunked JSON reader)
+            if self.reader_options["infer_schema_length"] != 100:
+                raise NotImplementedError(
+                    "infer_schema_length is not supported in the JSON reader"
+                )
```

Comment:
You should have a schema by now because I think it runs early?

Reply:
Thanks, removed this part!